pyquil/api/_qvm.py

```diff
@@ -565,17 +565,18 @@ def run(self) -> "QVM":
 
         quil_program = self.augment_program_with_memory_values(quil_program)
 
-        self._memory_results = self.connection._qvm_run(
+        for region in quil_program.declarations.keys():
+            self._memory_results[region] = np.ndarray((trials, 0), dtype=np.int64)
```

Comment:
I think this would be more appropriate in `load()`, and the region type should probably mirror that of the declaration.

Reply:
Ah yes, I remember that now. Good catch. I was taking advantage of the `if not isinstance(self._executable, Program):` above, but it probably makes most sense in `load()` either way (there may be an instance check on load too...need to look)

Reply:
Also, I believe we risk altering existing behavior by doing this (e.g. `DECLARE ro BIT` could change if we mapped to byte or bool instead of int64). Do we want to consider the existing behavior a bug? If so, I would think the second dimension should also match the declaration... Though not sure if we want to introduce that change too, as it adds to complexity for this issue.

Reply:
Per outside conversation, deciding to punt on type mapping at this time